central/networkgraph/aggregator/aggregator_impl.go

```diff
@@ -174,7 +175,17 @@ func (a *aggregateExternalConnByNameImpl) Aggregate(flows []*storage.NetworkFlow
 		}
 
 		flow = flow.CloneVT()
-		srcEntity, dstEntity = flow.GetProps().GetSrcEntity(), flow.GetProps().GetDstEntity()
+
+		flowProps := flow.GetProps()
+		if flowProps == nil {
+			continue
+		}
+
+		// Anonymize entities inside the flow as needed
+		flowProps.SrcEntity = anonymizeIfNoExternalIPs(flowProps.SrcEntity, features.NetworkGraphExternalIPs.Enabled())
+		flowProps.DstEntity = anonymizeIfNoExternalIPs(flowProps.DstEntity, features.NetworkGraphExternalIPs.Enabled())
+
+		srcEntity, dstEntity = flowProps.SrcEntity, flowProps.DstEntity
```

Comment:
Let's make it clear at this level, that the anonymization happens only if that feature is off.
```suggestion
		// Anonymize entities inside the flow as needed
		if !features.NetworkGraphExternalIPs.Enabled() {
			flowProps.SrcEntity = anonymizeIfNoExternalIPs(flowProps.SrcEntity, features.NetworkGraphExternalIPs.Enabled())
			flowProps.DstEntity = anonymizeIfNoExternalIPs(flowProps.DstEntity, features.NetworkGraphExternalIPs.Enabled())
			srcEntity, dstEntity = flowProps.SrcEntity, flowProps.DstEntity
		}

```

Reply:
Additionally, it would be cool to write why are we doing the anonymization. I know the reason, but let's make sure that the other readers of this code are aware as well.